Libs/MRML/Core/vtkMRMLTableSQLiteStorageNode.cxx

```diff
@@ -0,0 +1,269 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Andras Lasso (PerkLab, Queen's
+  University) and Kevin Wang (Princess Margaret Hospital, Toronto) and was
+  supported through OCAIRO and the Applied Cancer Research Unit program of
+  Cancer Care Ontario.
+
+==============================================================================*/
+
+// MRML includes
+#include "vtkMRMLTableSQLiteStorageNode.h"
+#include "vtkMRMLTableNode.h"
+#include "vtkMRMLScene.h"
+
+// VTK includes
+#include <vtkObjectFactory.h>
+#include <vtkTable.h>
+#include <vtkStringArray.h>
+#include <vtkBitArray.h>
+#include <vtkNew.h>
+#include <vtkSQLQuery.h>
+#include <vtkRowQueryToTable.h>
+#include <vtkSQLDatabase.h>
+#include <vtkSQLiteDatabase.h>
+#include <vtkSQLiteQuery.h>
+
+#include <vtksys/SystemTools.hxx>
+
+//------------------------------------------------------------------------------
+vtkMRMLNodeNewMacro(vtkMRMLTableSQLiteStorageNode);
+
+//----------------------------------------------------------------------------
+vtkMRMLTableSQLiteStorageNode::vtkMRMLTableSQLiteStorageNode()
+{
+  this->TableName = 0;
+  this->Password = 0;
+}
+
+//----------------------------------------------------------------------------
+vtkMRMLTableSQLiteStorageNode::~vtkMRMLTableSQLiteStorageNode()
+{
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLTableSQLiteStorageNode::PrintSelf(ostream& os, vtkIndent indent)
+{
+  vtkMRMLStorageNode::PrintSelf(os,indent);
+}
+
+//----------------------------------------------------------------------------
+bool vtkMRMLTableSQLiteStorageNode::CanReadInReferenceNode(vtkMRMLNode *refNode)
+{
+  return refNode->IsA("vtkMRMLTableNode");
+}
+
+//----------------------------------------------------------------------------
+int vtkMRMLTableSQLiteStorageNode::ReadDataInternal(vtkMRMLNode *refNode)
+{
+  std::string fullName = this->GetFullNameFromFileName();
+
+  if (fullName == std::string(""))
+    {
+    vtkErrorMacro("ReadData: File name not specified");
+    return 0;
+    }
+  vtkMRMLTableNode *tableNode = vtkMRMLTableNode::SafeDownCast(refNode);
+  if (tableNode == NULL)
+    {
+    vtkErrorMacro("ReadData: unable to cast input node " << refNode->GetID() << " to a table node");
+    return 0;
+    }
+
+  // Check that the file exists
+  if (vtksys::SystemTools::FileExists(fullName) == false)
+    {
+    vtkErrorMacro("ReadData: table file '" << fullName << "' not found.");
+    return 0;
+    }
+
+  std::string dbname = std::string("sqlite://") + fullName;
+  vtkSQLiteDatabase *database = vtkSQLiteDatabase::SafeDownCast( vtkSQLiteDatabase::CreateFromURL(dbname.c_str()));
```

Comment:
Use vtkSmartPointer::Take to avoid littering the code with Delete() statements late, similar to this:

vtkSmartPointer<vtkAbstractTransform> transformCopy=vtkSmartPointer<vtkAbstractTransform>::Take(transform->MakeTransform());